lang/py/src/avro/schema.py

```diff
@@ -405,6 +424,83 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale=0, size=0, fixed_schema=None):
+    if decimal_type != 'fixed' and decimal_type != 'bytes':
```

Comment:
I have revamped the way I designed these classes based on your feedback. This would look much cleaner.